app/handlers/newflow/educator_signup.rb

```diff
@@ -13,11 +23,83 @@ def authorized?
     end
 
     def required_params
-      @required_params ||= [:email, :first_name, :last_name, :password, :phone_number, :role]
+      @required_params ||= [:email, :first_name, :last_name, :password]
     end
 
     def handle
-      super
+      new_login_credential_check
+      return if errors?
+
+      outputs.email = signup_params.email
+
+      if LookupUsers.by_verified_email(signup_params.email).first
+        fatal_error(
+          code: :email_taken,
+          message: I18n.t(:"login_signup_form.email_address_taken"),
+          offending_inputs: :email
+        )
+      end
+
+      if options[:user_from_signed_params].present?
+        outputs.user = User.find_by!(id: options[:user_from_signed_params]['id'])
+      else
+        outputs.user = create_user
+
+        run(::SetPassword,
+          user: outputs.user,
+          password: signup_params.password,
+          password_confirmation: signup_params.password
+        )
+      end
+
+      agree_to_terms
+      run(CreateEmailForUser, email: signup_params.email, user: outputs.user)
+    end
+
+  private ###################
+
+    def new_login_credential_check
```

Comment:
are all these conditionals tested in unit tests?

Reply:
I don't think so. Not yet. This is code that you created btw 🙃 — it was your first PR, I think, in Accounts.